Sentinel/Classes/Core/Internal/SentinelInternal.swift

```diff
@@ -20,97 +20,168 @@ extension Sentinel {
         preferences: [OptionSwitchItem],
         on viewController: UIViewController
     ) {
-        let tabBarControllers = createTabBarControllers(
+        let tabItems = createTabItems(
             with: tools,
             preferences: preferences,
             viewController: viewController
         )
         let tabBarController = UIStoryboard.sentinel
             .instantiateViewController(ofType: SentinelTabBarController.self)
-        tabBarController.setupViewControllers(with: tabBarControllers)
+        let preselectedTabIndex = preselectedTabIndex(
+            for: .tools(items: []),
+            tabItems: tabItems
+        )
+        tabBarController.setupViewControllers(
+            with: tabItems.map { $0.viewController },
+            preselectedIndex: preselectedTabIndex
+        )
         tabBarController.title = "Sentinel"
 
         let navController = UINavigationController(rootViewController: tabBarController)
         viewController.present(navController, animated: true)
     }
 }
 
-// MARK: - Private extension
+enum Tab {
+    case device
+    case application
+    case tools(items: [Tool])
+    case preferences(items: [OptionSwitchItem])
+    case performance
+}
 
-private extension Sentinel {
-    func createTabBarControllers(
-        with tools: [Tool],
-        preferences: [OptionSwitchItem],
-        viewController: UIViewController
-    ) -> [UIViewController] {
-        return [
-            createDeviceViewController(),
-            createApplicationViewController(),
-            createToolsController(with: tools),
-            createPreferencesViewController(preferences),
-            createPerformanceViewController()
-        ]
-    }
+struct TabItem {
```

Comment:
Just the last thing that needs to be done - please add both `Tab` and `TabItem` into a new file (e.g. TabItem.swift).